R/describe-feature.R

```diff
@@ -0,0 +1,82 @@
+# Copyright 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+#' Describe a WFS feature
+#'
+#' Describe the columns from a WFS feature. The column name, whether a column can be
+#' separated from the record in WFS (nillable) and the tpye of column are returned.
+#' This can be a helpful tool to examine a layer before issuing a query with `bcdc_get_geodata`
+#'
+#' @inheritParams bcdc_get_geodata
+#' @export
+#'
+#' @examples
+#'  bcdc_describe_feature("bc-airports")
+
+bcdc_describe_feature <- function(x = NULL){
+  obj <- bcdc_get_record(x)
+  if (!"wms" %in% vapply(obj$resources, `[[`, "format", FUN.VALUE = character(1))) {
+    stop("No wms/wfs resource available for this dataset.",
+         call. = FALSE
+    )
+  }
+
+  ## Parameters for the API call
+  query_list <- list(
+    SERVICE = "WFS",
+    VERSION = "2.0.0",
+    REQUEST = "DescribeFeatureType",
+    #outputFormat = "application/json",
+    typeNames = obj$layer_name
+  )
+
+
+  ## GET and parse data to sf object
+  cli <- bcdc_http_client(url = "https://openmaps.gov.bc.ca/geo/pub/wfs")
+
+  cc <- cli$get(query = query_list)
+  status_failed <- cc$status_code >= 300
+
+  xml_res <- xml2::read_xml(cc$parse("UTF-8"))
+  xml_res <- xml2::xml_find_all(xml_res, "//xsd:sequence")
+  xml_res <- xml2::xml_find_all(xml_res, ".//xsd:element")
+  xml_res <-  purrr::map(xml_res, xml2::xml_attrs)
+  xml_df <- purrr::map_df(xml_res, ~as.list(.))
```

Comment:
Suggest we pull this logic out into a little function

Reply:
Sounds good. While I am in there, I also wonder if this function might be better called `glimpse` since it is relatively equivalent to what `dplyr::glimpse` does. Or maybe I'll leave it as is but define a glimpse method that draws from `describe-feature()`

Reply:
I like that idea.

On second thought, maybe we don't need another function at this point - it's all together in one place now, and doesn't need to be reused?